snap_plugin/v1/metric.py

```diff
@@ -0,0 +1,259 @@
+# -*- coding: utf-8 -*-
+# http://www.apache.org/licenses/LICENSE-2.0.txt
+#
+# Copyright 2016 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+from builtins import int
+
+from past.builtins import basestring
+
+from .config_map import ConfigMap
+from .namespace import Namespace
+from .plugin_pb2 import Metric as PbMetric
+from .timestamp import Timestamp
+
+
+class Metric(object):
+    """Metric
+
+    The metric object is the core entity of snap representing metrics.
+
+    Args:
+        namespace (:obj:`list` of
+            :py:class:`~snap_plugin.v1.namespace_element.NamespaceElement`):
+            namespace elements
+        version (:obj:`int`): metric version
+        tags (:obj:`dict`): metric tags (key/value pairs)
+        config (:py:class:`~snap_plugin.v1.config_map.ConfigMap`): config for
+            metric (key/value pairs)
+        timestamp (:obj:`float`): metric timestamp (see time.time())
+        unit (:obj:`str`): metric unit
+        description (:obj:`str`): metric description
+
+    """
+    def __init__(self, namespace=[], version=0, tags={}, config={},
+                 timestamp=time.time(), unit="", description="", **kwargs):
+        if "pb" in kwargs:
+            self._pb = kwargs.get("pb")
+            self._config = ConfigMap(pb=self._pb.Config)
+            self._namespace = Namespace(pb=self._pb.Namespace)
+            self._timestamp = Timestamp(pb=self._pb.Timestamp)
+            if self._pb.HasField("int32_data"):
+                self._data_type = int
+            elif self._pb.HasField("int64_data"):
+                self._data_type = int
+            elif self._pb.HasField("float64_data"):
+                self._data_type = float
+            elif self._pb.HasField("string_data"):
+                self._data_type = str
+            elif self._pb.HasField("bool_data"):
+                self._data_type = bool
+            elif self._pb.HasField("bytes_data"):
+                self._data_type = bytes
```

Comment:
Does this list need to be extended for https://github.com/intelsdi-x/snap/issues/1250 ?
(I think @candysmurf is working on that)